src/resources.rs

```diff
@@ -223,6 +227,51 @@ pub fn add_tcp_stream(stream: tokio::net::TcpStream) -> Resource {
   Resource { rid }
 }
 
+pub fn add_http_server(s: http_server::HttpServer) -> Resource {
+  let rid = new_rid();
+  let mut tg = RESOURCE_TABLE.lock().unwrap();
+  match tg.insert(rid, Repr::HttpServer(s)) {
+    Some(_) => panic!("There is already a file with that rid"),
+    None => Resource { rid },
+  }
+}
+
+pub fn add_http_transaction(transaction: http_server::Transaction) -> Resource {
+  let rid = new_rid();
+  let mut tg = RESOURCE_TABLE.lock().unwrap();
+  match tg.insert(rid, Repr::HttpTransaction(transaction)) {
+    Some(_) => panic!("There is already a file with that rid"),
+    None => Resource { rid },
+  }
+}
+
+pub fn http_accept(
+  rid: ResourceId,
+) -> impl Future<Item = http_server::Transaction, Error = DenoError> {
+  let mut table = RESOURCE_TABLE.lock().unwrap();
+  let maybe_repr = table.get_mut(&rid);
+  match maybe_repr {
+    Some(Repr::HttpServer(ref mut s)) => Either::A(s.accept()),
+    _ => Either::B(futures::future::err(bad_resource())),
+  }
+}
+
+pub fn http_write_response(
+  rid: ResourceId,
+  response: http_server::Res,
+) -> DenoResult<()> {
+  let mut table = RESOURCE_TABLE.lock().unwrap();
+  let maybe_repr = table.get_mut(&rid);
+  match maybe_repr {
+    Some(Repr::HttpTransaction(ref mut transaction)) => {
+      let tx = transaction.response_tx.take().unwrap();
+      tx.send(response).unwrap();
+      Ok(())
+    }
+    _ => Err(bad_resource()),
```

Comment:
If I am understanding correctly, double `res.writeResponse()` would trigger `bad_resource()` as the first call would take ownership. Just a bit worried about that `bad rid` does not sound like a very helpful error message if the user does so (probably we can improve the error message later on...)